src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/test_keyvault_commands.py

```diff
@@ -888,5 +888,46 @@ def test_keyvault_storage_account(self, resource_group):
         self.cmd('keyvault storage list --vault-name {kv}', checks=[self.check('length(@)', 0)])
 
 
+class KeyVaultNetworkRuleScenarioTest(ScenarioTest):
+
+    def _create_subnet(test, kwargs, additional_args=None):
+        test.cmd('network vnet create -g {rg} -n {vnet} -l {loc} ')
+        test.cmd('network vnet subnet create -g {rg} --vnet-name {vnet} --name {subnet} '
+                 '--address-prefixes 10.0.0.0/21 --service-endpoints Microsoft.KeyVault')
+        return test.cmd('network vnet subnet show -g {rg} --vnet-name {vnet} --name {subnet}')
+
+    @ResourceGroupPreparer(name_prefix='cli_test_keyvault_network_rule')
+    def test_keyvault_network_rule(self, resource_group):
+
+        self.kwargs.update({
+            'kv': self.create_random_name('cli-test-keyvault-', 24),
```

Comment:
like we discussed on race condition unique to key-vault tests, consider using a unique prefix

Reply:
All certs have expired. Need more time to fix them. Let's fix firewall issue first.